src/routes/genres.routes.ts

```diff
@@ -0,0 +1,156 @@
+import { Router, Request, Response, NextFunction } from 'express';
+
+import { GenreModel } from '../models/genres.schemas';
+import { Genre } from '../models/models';
+
+const router: Router = Router();
+
+/**
+ * @swagger
+ *    tags:
+ *    name: Genres
+ * /genres:
+ *   get:
+ *     summary: Get genres list
+ *     tags:
+ *       - Genres
+ *     responses:
+ *       200:
+ *         description: List of genres
+ *         content:
+ *           application/json:
+ *             example:
+ *               - name: Action
+ *               - name: Drama
+ *       404:
+ *         description: Genres not found
+ *       500:
+ *         description: Internal Server Error
+ */
+router.get('/', (req: Request, res: Response, next: NextFunction) => {
+  GenreModel.find()
+    .then((genres: Genre[]) => {
+      if (genres.length === 0) {
+        return res.status(200).json({ data: 'Genres not added yet' });
+      }
+
+      return res.status(200).json({ data: genres });
+    })
+    .catch(error => next(error));
+});
+
+/**
+ * @swagger
+ * /genres:
+ *   post:
+ *     summary: Create new genre
+ *     tags:
+ *       - Genres
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           example:
+ *             name: New genre
+ *     responses:
+ *       201:
+ *         description: Genre created successfully
+ *         content:
+ *           application/json:
+ *             example:
+ *               name: New genre
+ *       400:
+ *         description: Bad Request
+ *       500:
+ *         description: Internal Server Error
+ */
+router.post('/', (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Is it better to add validation middleware for endpoints (not 'GET') to validate each field that come in your paylaod, using for example Joi package or writting it by your own.

Reply:
used joi for validation